pkg/cmd/repo/sync/sync.go

```diff
@@ -256,7 +256,7 @@ func executeLocalRepoSync(srcRepo ghrepo.Interface, remote string, opts *SyncOpt
 	}
 	if currentBranch == branch {
 		if isDirty, err := git.IsDirty(); err == nil && isDirty {
-			return fmt.Errorf("can't sync because there are local changes; please stash them before trying again")
+			return fmt.Errorf("can't sync because there are local changes; please stash them before trying again. Use git stash --all for offending untracked files.")
```

Comment:
I'm not a fan of this error message since it can be potentially confusing to the user.

First, the `IsDirty()` method doesn't tell us what was "dirty" in the first place: was it a tracked file with uncommitted changes, an untracked file, or both? To answer that, we could in theory extend our git client with a more informative method based on [UncommittedChangeCount](https://github.com/cli/cli/blob/e5a0b1fe34e7e2b0b2db90379b894e93680fdfd2/git/client.go#L197), but that might require more work on this PR that you've bargained for, so for now let's maybe avoid that.

If an error message gives actionable instructions, we want those instructions to be as generic as possible and ideally work for everyone. The above error message essentially says to use stash or `stash --all` depending on the circumstances, but we have to keep in mind that the user doesn't necessarily understand the circumstances. The message should also ideally not be misleading, e.g. by suggesting that there might have been “offending untracked” files when there were none.

To address all that, I propose something like:
```suggestion
			return fmt.Errorf("refusing to sync due to uncommitted local changes.\nUse `git stash --all` before retrying sync and `git stash pop` afterwards")
```
In Go in general, this much sentence formatting in an error message is frowned upon, but since this error will be printed pretty much unchanged to stderr, multiline messages with sentences are fine when returned from the main command function itself.

Reply:
That should help. I added some extra phrasing around it. Should make it clearer. Please let me know if you prefer the more compact phrasing.